pwnlib/elf/corefile.py

```diff
@@ -429,6 +429,15 @@ class Corefile(ELF):
         -1
         >>> io.corefile.signal == signal.SIGTRAP # doctest: +SKIP
         True
+
+        Make sure fault_addr synthesis works for amd64 on ret.
+
+        >>> context.clear(arch='amd64')
+        >>> elf = ELF.from_assembly('push 1234; ret')
+        >>> io = elf.process()
+        >>> io.wait()
+        >>> io.corefile.fault_addr
+        1234
```

Comment:
This is not a good test if I understand the issue correctly. From my tests, it seems that `fault_addr` was only null before this patch if it would've been `>= 2**47`. This is because on 64 bit, that's where kernel memory is mapped so we get "protection fault" instead of "segfault". A protection fault does not have the same `siginfo` as a normal segfault: `si_code` will be `SI_KERNEL = 0x80` and there won't be a `fault_addr` (see http://elixir.free-electrons.com/linux/v4.14-rc8/source/kernel/signal.c#L1052)

Reply:
Fixed in #1031

Reply:
Err Fixed in #1066